core/transfer/src/main/java/org/eclipse/dataspaceconnector/transfer/core/transfer/TransferProcessManagerImpl.java

```diff
@@ -171,23 +174,33 @@ public void handleProvisionResult(String processId, List<ProvisionResponse> resp
 
         responses.stream()
                 .map(response -> {
-                    var destinationResource = response.getResource();
+                    var provisionedResource = response.getResource();
                     var secretToken = response.getSecretToken();
 
-                    if (destinationResource instanceof ProvisionedDataDestinationResource) {
-                        var dataDestinationResource = (ProvisionedDataDestinationResource) destinationResource;
-                        DataAddress dataDestination = dataDestinationResource.createDataDestination();
+                    if (provisionedResource instanceof ProvisionedDataDestinationResource) {
+                        // a data destination was provisioned by a consumer
+                        var dataDestinationResource = (ProvisionedDataDestinationResource) provisionedResource;
+                        var dataDestination = dataDestinationResource.createDataDestination();
 
                         if (secretToken != null) {
-                            String keyName = dataDestinationResource.getResourceName();
+                            var keyName = dataDestinationResource.getResourceName();
                             vault.storeSecret(keyName, typeManager.writeValueAsString(secretToken));
                             dataDestination.setKeyName(keyName);
                         }
-
                         transferProcess.getDataRequest().updateDestination(dataDestination);
+                    } else if (provisionedResource instanceof ProvisionedContentResource) {
+                        // content for the data transfer was provisioned by the provider
+                        var contentResource = (ProvisionedContentResource) provisionedResource;
+                        var contentAddress = contentResource.getContentDataAddress();
+                        if (secretToken != null) {
```

Comment:
Not yet because those methods aren't defined on a superclass. That could be looked into in a subsequent PR but I'm not sure it will buy that much in terms of commonality.